crates/wallet/src/wallet/coin_selection.rs

```diff
@@ -322,23 +322,22 @@ pub fn decide_change(remaining_amount: u64, fee_rate: FeeRate, drain_script: &Sc
 fn select_sorted_utxos(
     utxos: impl Iterator<Item = (bool, WeightedUtxo)>,
     fee_rate: FeeRate,
-    target_amount: u64,
+    target_amount: Amount,
     drain_script: &Script,
 ) -> Result<CoinSelectionResult, InsufficientFunds> {
-    let mut selected_amount = 0;
-    let mut fee_amount = 0;
+    let mut selected_amount = Amount::ZERO;
+    let mut fee_amount = Amount::ZERO;
     let selected = utxos
         .scan(
             (&mut selected_amount, &mut fee_amount),
             |(selected_amount, fee_amount), (must_use, weighted_utxo)| {
                 if must_use || **selected_amount < target_amount + **fee_amount {
-                    **fee_amount += (fee_rate
+                    **fee_amount += fee_rate
                         * (TxIn::default()
                             .segwit_weight()
                             .checked_add(weighted_utxo.satisfaction_weight)
-                            .expect("`Weight` addition should not cause an integer overflow")))
-                    .to_sat();
-                    **selected_amount += weighted_utxo.utxo.txout().value.to_sat();
+                            .expect("`Weight` addition should not cause an integer overflow"));
```

Comment:
nit: we can remove the extra round brackets starting at `TxIn::default()`. same suggestion on L381.

Reply:
Fixed.